viadot/flows/outlook_to_adls.py

```diff
@@ -0,0 +1,101 @@
+import os
+from typing import Any, Dict, List, Union, Literal
+
+import pendulum
+from prefect import Flow, Task, apply_map, task
+import pandas as pd
+from ..utils import slugify
+from ..task_utils import df_to_csv, union_dfs_task
+
+from ..tasks import OutlookToDF, AzureDataLakeUpload
+
+file_to_adls_task = AzureDataLakeUpload()
+outlook_to_df = OutlookToDF()
+
+
+class OutlookToADLS(Flow):
+    def __init__(
+        self,
+        mailbox_list: List[str],
+        name: str = None,
+        start_date: str = None,
+        end_date: str = None,
+        local_file_path: str = None,
+        extension_file: str = ".csv",
+        adls_file_path: str = None,
+        overwrite_adls: bool = True,
+        adls_sp_credentials_secret: str = None,
+        limit: int = 10000,
+        if_exists: Literal["append", "replace", "skip"] = "append",
+        *args: List[Any],
+        **kwargs: Dict[str, Any],
+    ):
+        """Flow for downloading data from Outlook source to a local CSV
+        using Outlook API, then uploading it to Azure Data Lake.
```

Comment:
Here you download data to DF, not to CSV file. Converting to csv file is part of uploading to Azure DL

Reply:
This is the only suggestion - maybe the description is well written.